tools/topology/sof/pipe-smart-amplifier-playback.m4

```diff
@@ -69,6 +69,16 @@ C_CONTROLBYTES(Smart_amp Model, PIPELINE_ID,
         ,
         MODEL_priv)
 
+
+ifelse(SOF_ABI_VERSION_3_17_OR_GRT, `1',
+C_CONTROLBYTES_VOLATILE_READONLY(Smart_amp Model_Get_params, PIPELINE_ID,
+        CONTROLBYTES_OPS(bytes, 258 binds the mixer control to bytes get handlers, 258, 258),
+        CONTROLBYTES_EXTOPS_READONLY(260 binds the mixer control to bytes get handlers, 260),
+        , , ,
+        CONTROLBYTES_MAX(, 300000),
+        ,
+        MODEL_priv))
+
```

Comment:
Sorry to chime in late, why we need an extra read-only kcontrol if it denotes the same model configure params? can't we just modify the existed one directly to be RW-volatile?

Reply:
@keyonjie we want to keep our RW control non-volatile so that what we write and subsequently read are consistent.

Reply:
> @keyonjie we want to keep our RW control non-volatile so that what we write and subsequently read are consistent.

How will the FW module use this model data, if the RW one and the Read only one are with the same meaning, we don't need to preserve the original written one consistent with what we have written. e.g. we write an volume level, and then FW update it internally, the read back of the volume should return this FW internal updated one.

Reply:
My point is adding one more kcontrol with almost the same meaning could confuse us.

Reply:
Hi Keyon,
The actual intention here is not to wake up DSP, if the intention of the user is just to read back the parameters that he has set. Eg: Hotword model.
Whereas this Read volatile control is actually going to return the real-time data, that will be helpful for the tuning purpose. In this way we can minimize the DSP wakes for reading the parameters.

Reply:
The parameters cached from that been set will mislead us, we should have only one original data (e.g. from real register or real-time data) which should be read back. In case that the cached data is different from the real-time one, the cached one makes no sense but only introducing confusion. IMHO we should delete this cached one but keep the real-time one only if it will be changed by the Firmware, reading back the misleading cached data to avoid waking up the DSP looks worse than refusing to reading it back when DSP is suspended.